### PR TITLE
Fix jetstream removeSub logic

### PR DESCRIPTION
**Pull request by boxboatmatt**
Signed-off-by: Matthew DeVenny <matt@boxboat.com>

When `removeSub` calls `jsi.hbc.Stop()` it is possible that the timer has already expired
https://github.com/nats-io/nats.go/blob/main/nats.go#L3919

from `timer.Stop()`
```
For a timer created with AfterFunc(d, f), if t.Stop returns false, then the timer has already expired 
and the function f has been started in its own goroutine; Stop does not wait for f to complete before 
returning. If the caller needs to know whether f is completed, it must coordinate with f explicitly.
```

This PR checks to see if `sub.closed == true` and returns

### CODE REVIEW

File: js.go [hunk 1]
```diff
@@ -1844,7 +1844,14 @@ func (sub *Subscription) activityCheck() {
 	}
 
 	active := jsi.active
-	jsi.hbc.Reset(jsi.hbi)
+	// the timer could still have gone off even though it was stopped in removeSub,
+	// in this case it will be nil because removeSub sets it to nil
+	if jsi.hbc != nil {
+		jsi.hbc.Reset(jsi.hbi)
+	} else {
+		active = false
```
Comment (boxboatmatt) [inline js.go:1852]: Not 100% sure whether we should be setting `active` or `closed` to false here in this dangling execution

File: js.go [hunk 2]
```diff
@@ -1844,7 +1844,14 @@ func (sub *Subscription) activityCheck() {
 	}
 
 	active := jsi.active
-	jsi.hbc.Reset(jsi.hbi)
+	// the timer could still have gone off even though it was stopped in removeSub,
```
Comment (kozlovic) [inline js.go:1847]: Would revert the changes here and instead few lines above change the condition to:
```
if jsi == nil || sub.closed {
    sub.mu.Unlock()
    return
}
```
(or maybe even `if sub.closed` would be enough)

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: js.go (hunk 2 above)
Comment (boxboatmatt) [reply js.go:1847]: Yup.. that would work

Comment (kozlovic) [reply js.go:1847]: Like I said, this change can be reverted. Since jsi.hbc is set to nil at the same time (and under sub.mu.Lock) we set sub.closed = true in removeSub, we are ok with original code.

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: js.go [hunk 3]
```diff
@@ -1838,7 +1838,7 @@ func (sub *Subscription) scheduleFlowControlResponse(reply string) {
 func (sub *Subscription) activityCheck() {
 	sub.mu.Lock()
 	jsi := sub.jsi
-	if jsi == nil {
+	if jsi == nil || sub.closed {
```
Comment (kozlovic) [inline js.go:1841]: Sorry about that.. So now, since here we exit if the subscription is closed, the capture down there where we do:
```
closed := sub.closed
sub.mu.Unlock()

if !active && !closed {
..
```
is silly... we can simply by removing `closed := sub.closed` and simply have `if !active {..`

**Review by kozlovic — CHANGES_REQUESTED** [review]

**Review by kozlovic — APPROVED** [review]
LGTM. Thank you for your contribution!
